src/spec.c

```diff
@@ -1272,10 +1290,6 @@ void IndexSpec_LegacyFree(void *spec) {
 
 void IndexSpec_Free(IndexSpec *spec) {
   if (spec->flags & Index_Temporary) {
-    // we are taking the index to a background thread to be released.
-    // before we do it we need to delete it from the index dictionary
-    // to prevent it from been freed again.c
-    dictDelete(specDict_g, spec->name);
```

Comment:
Why remove it?

Reply:
Since we rm_call `FT.DROPINDEX` and if we have already removed the index from the global dictionary here, we are going to fail.